File: core4/leetcode/Two_Pointers/N9/N9.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {

        int n = heights.size();

        int left = 0;
        int right = n - 1;
        int max_area = 0;

        while (left < right) {

            int width = right - left;
            int length = min(heights[left], heights[right]);
            int current_area = width * length;

            if (current_area > max_area) {
                max_area = current_area;
            }

            if (heights[left] < heights[right]) {
                left++;
            } else {
                right--;
            }
        }

        return max_area;
    }
};

File: core4/leetcode/Two_Pointers/N9/N9.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        
        n = len(heights)
        
        left = 0
        right = n - 1
        max_area = 0
        
        while left < right:
            width = right - left
            length = min(heights[right], heights[left])
            current_area = width * length
            
            max_area = max(max_area, current_area)
            
            if heights[left] < heights[right]:
                left += 1
            else:
                right -= 1
                
        return max_area
